app/src/hooks/useDownload.tsx

```diff
@@ -0,0 +1,53 @@
+import { useDialogContext } from './useContext';
+
+const useDownload = () => {
+  const dialogContext = useDialogContext();
+  /**
+   * handler for downloading raw data as JSON
+   * Note: currently this does not zip the file. Can be modified if needed.
+   *
+   * @param {any} data - to download
+   * @param {string} fileName - name of file excluding file extension ie: file1
+   */
+  const downloadJSON = (data: any, fileName: string) => {
+    const blob = new Blob([JSON.stringify(data, undefined, 2)], { type: 'application/json' });
+
+    const link = document.createElement('a');
+
+    link.download = `${fileName}.json`;
+
+    link.href = URL.createObjectURL(blob);
+
+    link.click();
+
+    URL.revokeObjectURL(link.href);
+  };
+
+  /**
+   * Downloads / views a signed url
+   * displays error dialog if signedUrlService throws error
+   * Note: allows a promise to be passed to handle different api services
+   *
+   * @async
+   * @param {SubmissionFeatureSignedUrlPayload} params
+   * @returns {Promise<[void]>}
+   */
+  const downloadSignedUrl = async (signedUrl: Promise<string> | string) => {
```

Comment:
nit: I think the @param and @returns are outdated.

Also, Id encourage using full grammar in jsdoc (complete sentences, periods/capitals, etc). Just helps with readability, and also makes the code look that much more complete/professional.

Reply:
Updated this (and the grammar). I'll use full grammar for jsdocs going forward.